src/emc/motion/motion.c

```diff
@@ -199,21 +199,24 @@ void reportError(const char *fmt, ...)
     va_end(args);
 }
 
-#ifndef va_copy
-#define va_copy(dest, src) ((dest)=(src))
-#endif
-
 static rtapi_msg_handler_t old_handler = NULL;
 static void emc_message_handler(msg_level_t level, const char *fmt, va_list ap)
 {
-    va_list apc;
-    va_copy(apc, ap);
-    if(level == RTAPI_MSG_ERR) emcmotErrorPutfv(emcmotError, fmt, apc);
-    if(old_handler) old_handler(level, fmt, ap);
-    va_end(apc);
+   if(level == RTAPI_MSG_ERR) {
+        va_list apc;
+        //va_start(ap,fmt) - prepared by calling function
+        va_copy(apc, ap);
+	emcmotErrorPutfv(emcmotError, fmt, apc);
+        va_end(apc);
+        va_end(ap);
```

Comment:
How will calling va_end(ap) affect the call to old_handler()?  As far as I can tell from rtapi/rtai_rtapi.c, old_handler() expect an active va_list variable, and closing calling va_end(ap) will break all calls to rtapi_msg_handler().